upgrade/php/ps_remove_controller_tab.php

```diff
@@ -0,0 +1,115 @@
+<?php
+
+function ps_remove_controller_tab($className, $quickAccessUrls = [])
+{
+    // select given tab
+    $tabsToBeDeleted = [];
+    $rolesToBeDeleted = [];
+    $tabToDelete = Db::getInstance()->getRow(
+        sprintf("SELECT id_tab, id_parent, class_name FROM %stab WHERE class_name = '%s'", _DB_PREFIX_, $className)
+    );
+
+    if (empty($tabToDelete)) {
+        return;
+    }
+
+    // get tabs and roles that should be deleted
+    getElementsToBeDeleted($tabToDelete['id_tab'], $tabToDelete['id_parent'], $className, $tabsToBeDeleted, $rolesToBeDeleted);
+
+    // delete tabs fetched by the recursive function
+    Db::getInstance()->execute(
+        sprintf(
+            'DELETE FROM %stab WHERE id_tab IN (%s)',
+            _DB_PREFIX_,
+            implode(', ', $tabsToBeDeleted)
+        )
+    );
+
+    // delete orphan tab langs
+    Db::getInstance()->execute(
+        sprintf(
+            'DELETE FROM `%stab_lang` WHERE `id_tab` NOT IN (SELECT id_tab FROM `%s_tab`)',
+            _DB_PREFIX_,
+            _DB_PREFIX_
+        )
+    );
+
+    // delete orphan legacy quick access links
+    $sqlLegacyQuickAccessLinkDeletion = sprintf(
+        "DELETE FROM `%squick_access_lang`
+        WHERE id_quick_access IN (SELECT id_quick_access FROM `%squick_access` WHERE link LIKE '%%controller=%s%%')",
+        _DB_PREFIX_,
+        _DB_PREFIX_,
+        $className
+    );
+
+    Db::getInstance()->execute($sqlLegacyQuickAccessLinkDeletion);
+    Db::getInstance()->execute(
+        sprintf(
+            "DELETE FROM `%squick_access` WHERE link LIKE '%%controller=%s%%'",
+            _DB_PREFIX_,
+            $className
+        )
+    );
+
+    if (!empty($quickAccessUrls)) {
+        // delete orphan quick access links (given links, for symfony urls)
+        foreach ($quickAccessUrls as &$link) {
+            $link = "'" . $link . "'";
+        }
+        Db::getInstance()->execute(
+            sprintf(
+                'DELETE FROM %squick_access WHERE link IN (%s)',
+                _DB_PREFIX_,
+                implode(', ', $quickAccessUrls)
+            )
+        );
+    }
+
+    // delete orphan roles
+    $sqlRoleDeletion = sprintf('DELETE FROM %sauthorization_role WHERE ', _DB_PREFIX_);
+    foreach ($rolesToBeDeleted as $key => $role) {
+        if ($key === 0) {
+            $sqlRoleDeletion .= "slug LIKE '" . $role . "'";
+            continue;
+        }
+        $sqlRoleDeletion .= "OR slug LIKE '" . $role . "'";
+    }
+    Db::getInstance()->execute($sqlRoleDeletion);
+}
+
+function getElementsToBeDeleted($idTab, $idParent, $className, &$tabsToBeDeleted, &$rolesToBeDeleted)
```

Comment:
I'm not a fan of passing values by reference but here it does the job 👍

Reply:
yes, me neither usually, but it's quite useful in the case of a recursive function, especially when we're not in a class context